File: bread.py
from sys import stdin,stdout

N = int(stdin.readline())
p1 = list(map(int,stdin.readline().split()))
p2 = [0 for i in range(N)]
second = list(map(int,stdin.readline().split()))
for i in range(N): #O(N)
    p2[p1[i]-1] = second[i]
seen = [False for _ in range(N)]; swaps = N
for i in range(N): #O(N)
    if not seen[i]:
        swaps -= 1
        j = i
        while not seen[j]:
            seen[j] = True
            j = p2[j] - 1
stdout.write("Impossible") if swaps%2 else stdout.write("Possible")
    
      

File: bread.java
import java.io.*;
import java.util.*;

public class bread {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        int N = Integer.parseInt(br.readLine());
        String[] firstLine = br.readLine().split(" "), secondLine = br.readLine().split(" ");
        int[] p1 = new int[N], p2 = new int[N], second = new int[N];
        for (int i = 0; i < N; i++) {
            p1[i] = Integer.parseInt(firstLine[i]);
            p2[i] = Integer.parseInt(secondLine[i]);
        }

        for (int i = 0; i < N; i++) second[p1[i] - 1] = p2[i];
        boolean[] seen = new boolean[N];
        int swaps = N;
        //O(N)
        for (int i = 0; i < N; i++) {
            if (!seen[i]) {
                swaps--;
                int j = i;
                while (!seen[j]) {
                    seen[j] = true;
                    j = second[j] - 1;
                }
            }
        }
        pw.print(swaps%2 == 1 ? "Impossible\n" : "Possible\n");
        br.close();
        pw.close();
    }
}
